flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/ChangelogKeyedStateBackend.java

```diff
@@ -609,6 +609,11 @@ private ChangelogSnapshotState completeRestore(
                 ((ChangelogState) changelogState).resetWritingMetaFlag();
             }
 
+            for (ChangelogKeyGroupedPriorityQueue<?> priorityQueueState :
+                    priorityQueueStatesByName.values()) {
+                priorityQueueState.resetWritingMetaFlag();
+            }
+
```

Comment:
I think this change should be covered by a test (I mentioned this [earlier](https://github.com/apache/flink/pull/17462#issuecomment-966958130)).
It was already missed in this PR.